app/scripts/lib/multichain-api/provider-request.js

```diff
@@ -18,38 +11,45 @@ export async function providerRequestHandler(
   end,
   hooks,
 ) {
-  const [scope, wrappedRequest] = Array.isArray(request.params)
-    ? request.params
-    : paramsToArray(request.params);
+  const { scope, request: wrappedRequest } = request.params;
 
-  if (!hooks.hasPermission(request.origin, Caip25EndowmentPermissionName)) {
+  const caveat = hooks.getCaveat(
+    request.origin,
+    Caip25EndowmentPermissionName,
+    Caip25CaveatType,
+  );
+  if (!caveat) {
     return end(new Error('missing CAIP-25 endowment'));
   }
 
-  const chainId = scope.split(':')[1];
+  // TODO: consider case when scope is defined in requireScopes and optionalScopes
+  const scopeObject =
+    caveat.value.requiredScopes[scope] || caveat.value.optionalScopes[scope];
+
+  if (!scopeObject) {
+    return end(new Error('unauthorized (scopeObject missing)'));
+  }
 
-  if (!chainId) {
-    return end(new Error('missing chainId'));
+  if (!scopeObject.methods.includes(wrappedRequest.method)) {
+    return end(new Error('unauthorized (method missing in scopeObject)'));
   }
 
+  // Do we need to try catch this?
+  const { reference } = parseCaipChainId(scope);
```

Comment:
yes for sure, unless we're validating the CAIP-2 scope somewhere above?

Reply:
ahh dumb question. When i wrote this i was thinking it wasn't coming from the request for some reason

Reply:
resolved here https://github.com/MetaMask/metamask-extension/pull/25589/commits/3c687d8fa11126338db1abdc7e5dc588180c6f5a